JobScheduler.js

```diff
@@ -147,26 +148,36 @@ class JobScheduler {
             }
             return resolve();
           } else {
-            logger.info('+-> System is in maintenance or last maintenance operation has failed :', _.pick(maintenanceInfo, ['progress', 'state', 'completedAt', 'reason', 'toVersion', 'fromVersion', 'createdAt', 'updatedAt', 'createdBy', 'updatedBy']));
-            const currTime = moment();
-            const maintenanceStartTime = _.get(maintenanceInfo, 'createdAt');
-            if (maintenanceStartTime && currTime.diff(maintenanceStartTime) > config.scheduler.maintenance_mode_time_out) {
-              logger.warn(`System in maintenance since ${maintenanceInfo.createdAt}. Exceeds configured maintenance timeout. Flagging the current maintenance window as aborted.`);
-              const success = resolve;
-              return maintenanceManager
-                .updateMaintenace(`System in maintenance beyond configured timeout time ${config.scheduler.maintenance_mode_time_out/1000/60} (mins)`,
-                  CONST.OPERATION.ABORTED,
-                  CONST.SYSTEM_USER)
-                .then(success)
-                .catch((err) => {
-                  logger.error('error occurred while updating maintenance info', err);
-                  reject(err);
-                })
-                .finally(() => {
-                  if (this.intervalTimer) {
+            const maintInfoAttrs = ['progress', 'state', 'completedAt', 'reason', 'toVersion', 'fromVersion', 'createdAt', 'updatedAt', 'createdBy', 'updatedBy'];
```

Comment:
is it better to put constants in dedicated constants.js file?

Reply:
These are not constants, but rather attribs that we want to pick from the model object. Same logger was referring same set of attribs. THe reason to put stuff in constants is also in case they are reusable. Should not bloat constants file with everything in there.